services/registration/controller/controller.go

```diff
@@ -257,6 +258,71 @@ func UpdateCurrentUserRegistration(w http.ResponseWriter, r *http.Request) {
 	json.NewEncoder(w).Encode(updated_registration)
 }
 
+/*
+	Endpoint to patch user registration attributes
+*/
+func PatchCurrentUserRegistration(w http.ResponseWriter, r *http.Request) {
+	id := r.Header.Get("HackIllinois-Identity")
+
+	if id == "" {
+		errors.WriteError(w, r, errors.MalformedRequestError("Must provide id in request.", "Must provide id in request."))
+		return
+	}
+
+	user_registration := datastore.NewDataStore(config.REGISTRATION_DEFINITION)
+	err := json.NewDecoder(r.Body).Decode(&user_registration)
+
+	if err != nil {
+		errors.WriteError(w, r, errors.InternalError(err.Error(), "Could not decode user registration information. Possible failure in JSON validation, or invalid registration format."))
+		return
+	}
+
+	user_registration.Data["id"] = id
+
+	user_info, err := service.GetUserInfo(id)
+
+	if err != nil {
+		errors.WriteError(w, r, errors.InternalError(err.Error(), "Could not get user info."))
+		return
+	}
+
+	original_registration, err := service.GetUserRegistration(id)
+
+	if err != nil {
+		errors.WriteError(w, r, errors.DatabaseError(err.Error(), "Could not get user's original registration."))
+		return
+	}
+
+	user_registration.Data["github"] = user_info.Username
+
+	user_registration.Data["createdAt"] = original_registration.Data["createdAt"]
```

Comment:
We shouldn't need to re-insert the "github" or "createdAt" fields. We only need to change the "updatedAt" field + the fields the user provided in the request.